platform/darwin/src/NSCompoundPredicate+MGLAdditions.mm

```diff
@@ -0,0 +1,51 @@
+#import "NSCompoundPredicate+MGLAdditions.h"
+
+#import "NSPredicate+MGLAdditions.h"
+#import "NSExpression+MGLAdditions.h"
+
+class PredicateEvaluator {
+public:
+    
+    std::vector<mbgl::style::Filter>getFilters (NSArray<NSPredicate *> *predicates) {
+        std::vector<mbgl::style::Filter>filters;
+        for (NSPredicate *predicate in predicates) {
+            filters.push_back(predicate.mgl_filter);
+        }
+        return filters;
+    }
+    
+    mbgl::style::Filter operator()(NSCompoundPredicate *predicate) {
```

Comment:
This function should be inlined into `-[NSCompoundPredicate(MGLAdditions) mgl_filter]`.